src/code/z_play.c

```diff
@@ -396,12 +396,11 @@ void Play_Init(GameState* thisx) {
 
     osSyncPrintf("ZELDA ALLOC SIZE=%x\n", THA_GetRemaining(&this->state.tha));
     zAllocSize = THA_GetRemaining(&this->state.tha);
-    zAlloc = (u32)GameState_Alloc(&this->state, zAllocSize, "../z_play.c", 2918);
+    zAlloc = (uintptr_t)GameState_Alloc(&this->state, zAllocSize, "../z_play.c", 2918);
     zAllocAligned = (zAlloc + 8) & ~0xF;
-    ZeldaArena_Init((void*)zAllocAligned, zAllocSize - zAllocAligned + zAlloc);
+    ZeldaArena_Init((void*)zAllocAligned, zAllocSize - (zAllocAligned - zAlloc));
     // "Zelda Heap"
-    osSyncPrintf("ゼルダヒープ %08x-%08x\n", zAllocAligned,
-                 (s32)(zAllocAligned + zAllocSize) - (s32)(zAllocAligned - zAlloc));
+    osSyncPrintf("ゼルダヒープ %08x-%08x\n", zAllocAligned, zAllocAligned + zAllocSize - (s32)(zAllocAligned - zAlloc));
```

Comment:
Not same but same pattern, the uintptr_t subtraction is cast to s32 here

Reply:
I think this is fine to keep as s32 since it's the alignment difference and should never be more than a few bytes, but sure if we had a signed size_t it might make sense to use it here

Reply:
the standard types are not just about bit width but defined as "can hold a value representing <x> and <list of uses> will work", though I agree this can stay s32 since the code deeply assumes the n64 architecture anyway

I mostly pointed this out in contrast to the other 2 occurrences